app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,20 @@
+class SessionsController < ApplicationController
+  skip_before_filter :verify_authenticity_token
+
+  def new
+    provider = params[:format]
+    redirect_to "/auth/#{provider}"
+  end
+
+  def create
+    auth_hash = request.env['omniauth.auth']
+    @spy = Spy.find_or_create_from_omniauth(auth_hash)
+    session[:spy_id] = @spy.id
+    redirect_to root_path
+  end
+
+  def destroy
+    reset_session
```

Comment:
I didn't know about ```reset_session```! It looks like the method was [deprecated back in Rails 2.3.8](http://apidock.com/rails/ActionController/Base/reset_session) though. Can you explain why you chose to use it here?